pkg/background/update_request_controller.go

```diff
@@ -295,34 +327,23 @@ func (c *controller) deletePolicy(obj interface{}) {
 			kyvernov1beta1.URGeneratePolicyLabel: p.Name,
 		}))
 
-		urList, err := c.urLister.List(selector)
-		if err != nil {
-			logger.Error(err, "failed to get update request for the resource", "label", kyvernov1beta1.URGeneratePolicyLabel)
-			return
-		}
-
-		for _, ur := range urList {
-			for _, generatedResource := range ur.Status.GeneratedResources {
-				logger.V(4).Info("retaining resource", "apiVersion", generatedResource.APIVersion, "kind", generatedResource.Kind, "name", generatedResource.Name, "namespace", generatedResource.Namespace)
-			}
-		}
-
 		if !generatePolicyWithClone {
-			urs, err := c.urLister.GetUpdateRequestsForClusterPolicy(p.Name)
-			if err != nil {
-				logger.Error(err, "failed to update request for the policy", "name", p.Name)
-				return
-			}
 			// re-evaluate the UR as the policy was updated
 			for _, ur := range urs {
 				logger.V(4).Info("enqueue the ur for cleanup", "ur name", ur.Name)
 				c.enqueueUpdateRequest(ur)
 			}
-		}
-		// re-evaluate the UR as the policy was updated
-		for _, ur := range urs {
-			logger.V(4).Info("enqueue the ur for cleanup", "ur name", ur.Name)
-			c.enqueueUpdateRequest(ur)
+		} else {
+			urList, err := c.urLister.List(selector)
+			if err != nil {
+				logger.Error(err, "failed to get update request for the resource", "label", kyvernov1beta1.URGeneratePolicyLabel)
+				return
+			}
+			for _, ur := range urList {
+				for _, generatedResource := range ur.Status.GeneratedResources {
+					logger.V(4).Info("retaining resource for cloned policy", "apiVersion", generatedResource.APIVersion, "kind", generatedResource.Kind, "name", generatedResource.Name, "namespace", generatedResource.Namespace)
```

Comment:
we are just logging here ?

Reply:
Yes, its just a information not the err

Reply:
Ok, wanted to make sure we didn't forget to enqueue something.